nzshm_common/util/compression.py

```diff
@@ -0,0 +1,43 @@
+"""Useful utility & helpers."""
+import base64
+import io
+import zipfile
+from pathlib import Path, PurePath
+from typing import Union
+
+
+def compress_path(source_path: str, output_zip: Union[str, PurePath]) -> str:
+    """Zip contents of source path and return the full archive path.
```

Comment:
If by "full path" you mean "absolute path", that's not guaranteed because if `output_zip` is a string we simply return that.